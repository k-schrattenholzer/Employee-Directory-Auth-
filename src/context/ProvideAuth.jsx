import { createContext, useContext, useMemo, useState } from 'react';
import { getUser } from '../services/users.js';

const ProvideAuth = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <ProvideAuth.Provider value={value}>{children}</ProvideAuth.Provider>;
};

const useUser = () => {
  const context = useContext(ProvideAuth);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { ProvideAuth, UserProvider, useUser };
