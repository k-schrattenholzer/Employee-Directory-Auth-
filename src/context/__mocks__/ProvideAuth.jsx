import { createContext, useContext, useMemo, useState } from 'react';


const ProvideAuth = createContext();

const UserProvider = ({ mockUser, children }) => {

  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : {}
  );

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <ProvideAuth.Provider value={value}>{children}</ProvideAuth.Provider>;
};

const useUser = () => {
  const context = useContext(ProvideAuth);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

export { UserProvider, useUser }
