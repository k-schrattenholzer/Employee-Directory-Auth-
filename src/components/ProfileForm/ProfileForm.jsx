import { useState } from 'react';
import { useUser } from '../../context/ProvideAuth.jsx';
import { createProfile, updateProfile } from '../../services/profiles.js';
import { useForm } from '../../hooks/useForm.js'

export default function ProfileForm({ isEditing = false }) {
  const { user } = useUser();

  const { formState, formError, handleFormChange, setFormError } = useForm({
    name: '',
    email: user.email,
    bio: '',
    birthday: '',
  });

  console.log(formState);

  const handleSubmit = async (e) => {
    e.preventDefault();

    const { name, email, bio, birthday } = formState;

    try {
      if(isEditing) {
        updateProfile({ name, email, bio, birthday })       
      } else {
        await createProfile({ name, email, bio, birthday });
        debugger;
      }
    } catch (error) {
      setFormError(error);
    }
  };

  return (
    <div>
      {/* {formError && <p>{formError}</p> } */}
      <fieldset>
        <legend>{isEditing ? 'Update Your Profile' : 'Set Up Your Profile'}</legend>
        <form onSubmit={handleSubmit}>
          <label htmlFor="name">Name</label>
          <input
            name="name"
            value={formState.name}
            required
            onChange={handleFormChange}
          />

          <label htmlFor="email">Email</label>
          <p>{user.email}</p>

          <label htmlFor="birthday">Birthday</label>
          <input
            name="birthday"
            type='date'
            value={formState.birthday}
            required
            onChange={handleFormChange}
          />

          <label htmlFor="bio">Tell us your secrets.</label>
          <textarea
            name="bio"
            id="bio"
            value={formState.bio}
            onChange={handleFormChange}
          />

          <button type='submit'>Done-zo</button>
        </form>
      </fieldset>
    </div>
  );
}
