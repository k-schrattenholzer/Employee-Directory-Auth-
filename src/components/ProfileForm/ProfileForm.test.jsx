import { render } from '@testing-library/react';
import { UserProvider } from '../../context/ProvideAuth.jsx';
import ProfileForm from './ProfileForm.jsx'

jest.mock('../../context/ProvideAuth.jsx');
jest.mock('../../services/profiles.js')
jest.mock('../../services/users.js');

it('should display a profile form', () => {
  const { container } =  render(
        <UserProvider>
          <ProfileForm />
        </UserProvider>
  )
  expect(container).toMatchSnapshot();
})