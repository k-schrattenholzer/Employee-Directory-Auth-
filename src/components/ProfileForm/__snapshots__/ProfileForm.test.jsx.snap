// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a profile form 1`] = `
<div>
  <div>
    
    <fieldset>
      <legend>
        Set Up Your Profile
      </legend>
      <form>
        <label
          for="name"
        >
          Name
        </label>
        <input
          name="name"
          required=""
          value=""
        />
        <label
          for="email"
        >
          Email
        </label>
        <p />
        <label
          for="birthday"
        >
          Birthday
        </label>
        <input
          name="birthday"
          required=""
          type="date"
          value=""
        />
        <label
          for="bio"
        >
          Tell us your secrets.
        </label>
        <textarea
          id="bio"
          name="bio"
        />
        <button
          type="submit"
        >
          Done-zo
        </button>
      </form>
    </fieldset>
  </div>
</div>
`;
